_playwright-tests/UI/CustomRepoCRUD.spec.ts

```diff
@@ -0,0 +1,97 @@
+import { test, expect } from '@playwright/test';
+import { navigateToRepositories } from './helpers/navHelpers';
+import { deleteAllRepos } from './helpers/deleteRepositories';
+import { closePopupsIfExist, getRowByNameOrUrl } from './helpers/helpers';
+
+export const repoNamePrefix = 'Repo-CRUD';
+export const randomName = () => (Math.random() + 1).toString(36).substring(2, 6);
+export const repoName = `${repoNamePrefix}-${randomName()}`;
+export const rank = () => Math.floor(Math.random() * 10 + 1).toString();
+export const randomNum = () =>
+  Math.floor(Math.random() * 10 + 1)
+    .toString()
+    .padStart(2, '0');
+
+export const url = `https://stephenw.fedorapeople.org/multirepos/${rank()}/repo${randomNum()}/`;
+
+test.describe('Custom Repositories CRUD', () => {
+  test('Add, Read, update, delete a repo', async ({ page }) => {
+    await test.step('Delete any CRUD test repos that exist', async () => {
+      await navigateToRepositories(page);
+      await closePopupsIfExist(page);
+      await deleteAllRepos(page, `&search=${repoNamePrefix}`);
+    });
+    await test.step('Create a repository', async () => {
+      // Click on the 'Add repositories' button
+      await navigateToRepositories(page);
+      await closePopupsIfExist(page);
+      await page.getByRole('button', { name: 'Add repositories' }).first().click();
+      const repositoryModal = page.locator('div[id^="pf-modal-part"]').first();
+      await expect(repositoryModal).toBeVisible();
+      await expect(page.locator('text="Add custom repositories"')).toBeVisible();
```

Comment:
The modal selector is vague - it just says "a modal is visible". So I want to say it is pointless. Await on line 31 actually checks that there is this text available in the modal, so that part is the actual check.
So I suggest we don't need lines 23 and 30.

Also why have you decided to use `locator('text="Add custom repositories"')` instead of https://playwright.dev/docs/locators#locate-by-text ?

Reply:
Hi, I don't like the type of locator on line 29 as it can change and is not clear in its meaning.
I'll remove line 29 and update 31 as your prefer.

Reply:
I had better idea, I extended the repositoryModal to include the title text.